### DISCUSSION TITLE
Upload History

### DISCUSSION THREAD

**Discussion by I-I-IT** [ente-io/ente#650] [Enhancements]
It will be useful to have an **history of the folders that have been uploaded** to know what remain to be uploaded. Will be useful when you have a dozen Google Takeout files. **What** do users and devs **think** about this ?

**Comment by vishnukvmd** [comment]
Hey, just to be clear, if you re-upload data, it'll skip through already uploaded files and process just the rest. So as a user, you don't have to keep a track of pending items.

**Comment by luleyleo** [comment]
I feel like sometimes Ente uploads photos again, after they have been re-/moved. Today I moved a bunch of photos between albums on my desktop and later, when I opened the mobile app, it started to upload 200 photos, which I definitely didn't shoot today. There is also the issue when clearing the trash without syncing on mobile first. An upload history could make such incidents easier to understand.

**Reply by ua741** [reply]
Are you using an iOS device? We have noticed that iOS sometimes changes either the internal metadata or the EXIF data of the file, resulting in the re-processing of the same file. If it's just internal metadata that Apple maintains for the photo, no actual re-upload happens because the file hash doesn't change. But in the other case, we end up re-uploading these files due to a change in the hash.

--

When you trash/delete a photo from desktop, and the photo is still present on the device, there are known cases when it will result in re-upload of such files. Relevant issue https://github.com/ente-io/ente/issues/1159

**Reply by luleyleo** [reply]
I'm using Android. And actually, I think it didn't _really_ re-upload anything, like you said. But having no upload history makes it difficult to ensure that this is the case.

**Comment by luleyleo** [comment]
Another use-case for this: When sharing an album, it would be nice to see when another person uploaded some photos to it.
